### PR TITLE
bugfix/gwe-77-sold-stock-count

### PR DESCRIPTION
**Pull request by joshxfi**
before:
image

after accept order:
image

### CODE REVIEW

File: src/components/Dashboard/Seller/Orders/OrderProducts.tsx [hunk 1]
```diff
@@ -15,6 +16,18 @@ const OrderProducts = ({ order }: { order: OrderItem }) => {
...
     });
+
+    const { increment } = firebase.firestore.FieldValue;
+
+    products.forEach(async (product) => {
+      await db
+        .collection("products")
+        .doc(product.id)
+        .update({
+          stock: increment(-1),
```
Comment (princejoogie) [inline src/components/Dashboard/Seller/Orders/OrderProducts.tsx:27]: shouldnt we decrement by number of quantity of the product? @joshxfi

**Review by princejoogie — CHANGES_REQUESTED** [review]

File: src/components/Dashboard/Seller/Orders/OrderProducts.tsx (hunk 1 above)
Comment (princejoogie) [reply src/components/Dashboard/Seller/Orders/OrderProducts.tsx:27]: e.g. i put an order for 5 laptops, this should decrement 5 from stock and add 5 to sold

Comment (joshxfi) [reply src/components/Dashboard/Seller/Orders/OrderProducts.tsx:27]: ohh yea, mbmb

**Comment by joshxfi** [comment]
I'll push a quick bugfix in a sec

**Comment by zomeru** [comment]
You also need to decrease the stock when an order has been made. And if the the seller didn't accept or the buyer cancels the order, it increases also by the count of items that has been ordered.

**Comment by zomeru** [comment]
> You also need to decrease the stock when an order has been made. And if the the seller didn't accept or the buyer cancels the order, it increases also by the count of items that has been ordered.

It's a small bug that I found when I was working with checkout.

**Comment by joshxfi** [comment]
> You also need to decrease the stock when an order has been made. And if the the seller didn't accept or the buyer cancels the order, it increases also by the count of items that has been ordered.

Yes that's what this pr is about. But we only decrease stock and increase sold when seller accepts order & buyer won't be able to cancel after the seller accepts (currently)

**Comment by joshxfi** [comment]
It might be redundant to increment on order since seller might not accept order or buyer cancels before seller accepts their order.

**Comment by princejoogie** [comment]
> > You also need to decrease the stock when an order has been made. And if the the seller didn't accept or the buyer cancels the order, it increases also by the count of items that has been ordered.
> 
> Yes that's what this pr is about. But we only decrease stock and increase sold when seller accepts order.

@joshxfi shouldnt you put the decrease of stock when the seller accepts the order? which is called here
https://github.com/princejoogie/ground-work-ecom/blob/ae5488edc40c30ca3a79bce79c6540320f577ece/src/components/Dashboard/Seller/Orders/OrderProducts.tsx#L14-L18

**Comment by joshxfi** [comment]
@princejoogie I already made that commit

**Comment by princejoogie** [comment]
> @princejoogie I already made that commit [f009179](https://github.com/princejoogie/ground-work-ecom/commit/f00917905dfaab562cd6b0338682a0ab00de4d14)

no more changes @joshxfi ?

**Comment by joshxfi** [comment]
@princejoogie I'll look/fix for some other bugs on next pr

**Pull request merged**